htsget_server/indexing.py

```diff
@@ -128,19 +128,22 @@ def calculate_stats(obj_id):
     if "access_methods" in drs_json:
         # if there are access methods, it's a file object
         file_obj = drs_operations._get_file_path(drs_json["id"])
-        if file_obj["checksum"] is None:
-            logger.debug(f"calculating checksum for {drs_json['id']}")
-            checksum = []
-            with open(file_obj["path"], "rb") as f:
-                bytes = f.read()  # read file as bytes
-                checksum = [{
-                    "type": "sha-256",
-                    "checksum": hashlib.sha256(bytes).hexdigest()
-                }]
-            logger.debug(f"done calculating checksum for {drs_json['id']}")
-            drs_json["checksums"] = checksum
-        else:
-            drs_json["checksums"] = [file_obj["checksum"]]
+        ## This block is commented out because it takes too much resource and currently isn't useful, see DIG-1718
+        # if file_obj["checksum"] is None:
+        #     logger.debug(f"calculating checksum for {drs_json['id']}")
+        #     checksum = []
+        #     with open(file_obj["path"], "rb") as f:
+        #         bytes = f.read()  # read file as bytes
+        #         checksum = [{
+        #             "type": "sha-256",
+        #             "checksum": hashlib.sha256(bytes).hexdigest()
+        #         }]
+        #     logger.debug(f"done calculating checksum for {drs_json['id']}")
+        #     drs_json["checksums"] = checksum
+        # else:
+        #     drs_json["checksums"] = [file_obj["checksum"]]
+        ## Still save the checksum in case someone actually ingested the checksum
+        drs_json["checksums"] = [file_obj["checksum"]]
```

Comment:
but won't this raise an exception if there isn't already a checksum? It should be in a conditional on "checksum" existing.

Reply:
I am not really sure what you mean and which it should be conditional on, the original code says: 
```
if file_obj["checksum"] is None:
```
So I assumed file_obj always has the "checksum" key (otherwise this would raise an exception if the key doesn't exist)
The way I wrote the code, if file_obj["checksum"] is None it will get put in the drs_object as None, if it is not None it will save the value. Is it possible for the "checksum" key to not exist in either the `drs_object` or `file_obj`?

Reply:
It's possible that I was wrong the first time and that we should check for its existence either way. I'd put it in; it certainly does not hurt to make sure that "checksum" exists in file_obj.

Reply:
so should the drs object always have a checksums? like if there is no checksum in file_obj, should I add drs_json["checksums"]=[None]?